caddyhttp/fastcgi/setup.go

```diff
@@ -55,26 +56,21 @@ func fastcgiParse(c *caddy.Controller) ([]Rule, error) {
 
 		args := c.RemainingArgs()
 
-		switch len(args) {
-		case 0:
+		if !(len(args) > 1) {
 			return rules, c.ArgErr()
-		case 1:
-			rule.Path = "/"
-			rule.Address = args[0]
-		case 2:
-			rule.Path = args[0]
-			rule.Address = args[1]
-		case 3:
-			rule.Path = args[0]
-			rule.Address = args[1]
-			err := fastcgiPreset(args[2], &rule)
-			if err != nil {
-				return rules, c.Err("Invalid fastcgi rule preset '" + args[2] + "'")
-			}
 		}
 
-		network, address := parseAddress(rule.Address)
-		rule.dialer = basicDialer{network: network, address: address}
+		rule.Path = args[0]
+		lastIndex := len(args)
+
+		// Last argument is preset
+		if err := fastcgiPreset(args[len(args)-1], &rule); err == nil {
+			lastIndex--
+		}
```

Comment:
Hmm, I agree, this could use some improvement. What if the user has a local (internal) hostname of `php` because that's what they run their PHP app on?